pkg/imager/imager.go

```diff
@@ -342,3 +429,62 @@ func (i *Imager) buildUKI(ctx context.Context, report *reporter.Reporter) error
 
 	return nil
 }
+
+type overlayInstallerExecutor struct {
+	commandPath string
+}
+
+func (o *overlayInstallerExecutor) GetOptions(extra overlay.InstallExtraOptions) (overlay.Options, error) {
+	cmdArgs := []string{
+		"get-options",
+	}
+
+	// parse extra as yaml
+	extraYAML, err := yaml.Marshal(extra)
+	if err != nil {
+		return overlay.Options{}, fmt.Errorf("failed to marshal extra: %w", err)
+	}
+
+	cmd := exec.Command(o.commandPath, cmdArgs...)
+	cmd.Stdin = bytes.NewReader(extraYAML)
+
+	cmd.Stderr = os.Stderr
```

Comment:
we should capture `stderr` I guess and report it as an error, but also we can tidy up things a bit by having a more generic command executor which accepts an arg, stdin, and optionally unmarshals stdout, handles the error, etc.